expyfun/stimuli/_hrtf.py

```diff
@@ -49,20 +62,56 @@ def _get_hrtf(angle, source, fs):
     if angle < 0:
         leftward = True
         read_angle = -angle
-    if read_angle not in angles:
+    if read_angle not in angles and not interp:
         raise ValueError('angle "{0}" must be one of +/-{1}'
                          ''.format(angle, list(angles)))
     brir = data['brir']
-    idx = np.where(angles == read_angle)[0]
-    assert len(idx) == 1
-    brir = brir[idx[0]].copy()
+    if read_angle in angles:
+        interp = False
+    if not interp:
+        idx = np.where(angles == read_angle)[0]
+        assert len(idx) == 1
+        brir = brir[idx[0]].copy()
+    else:
+        idx = np.where(angles < read_angle)[0]
+        if (idx[-1] + 1) > (len(angles) - 1):
+            raise ValueError('angle "{0}" must be smaller than "{1}"'
+                             ''.format(angle, max(angles)))
+        # angles are
+        knowns = [angles[idx[-1]], angles[idx[-1] + 1]]
+        # get known brirs
+        brirs = [brir[idx[-1]].copy(), brir[idx[-1] + 1].copy()]
+
+        # find location of maximum for each component of each known brir and
+        # average to find a generally good place to declare time = 0 (all data
+        # from before this point gets shifted to the end)
+        delay = int(np.mean([np.argmax(brirs, 1)]))
```

Comment:
I did a bit of messing around with this today and I don't think the delay bit here makes a difference.